lib/fputs.js

```diff
@@ -0,0 +1,185 @@
+/**
```

Comment:
so I've thought about this, and my approach is to validate input that might go unnoticed,
but caller errors that would throw node errors anyway I delegate to node to report.
Especially for an argument that is guaranteed to be used on every codepath (the callback),
the test would be redundant.